application/cms/models.py

```diff
@@ -858,7 +866,7 @@ class DimensionClassification(db.Model):
 class ChartAndTableMixin(object):
 
     id = db.Column(db.Integer, primary_key=True)
-    classification_id = db.Column("classification_id", db.Integer)
+    classification_id = db.Column("classification_id", db.Integer, nullable=False)
```

Comment:
I'm not totally sure we want this to be non-nullable?  As it stands - yes - this table is only used for holding classifications, so if there's a row then this should exist.  But once we refactor the chart and table data out into this table will it still be the case we should only be able to save a row if it has a classification?

And if we **are** going to make this non-nullable then the three `includes_xxx` columns should probably also be non-nullable, as we need a classification plus those booleans to define a categorisation for the chart/table.  Just having the ID but not the booleans isn't right I think.

Reply:
Have made a few more things not nullable...

Reply:
We can easily make them nullable again if it comes to it.